src/Orleans.Core/Configuration/OptionLogger/DefaultOptionsFormatter.cs

```diff
@@ -1,104 +1,124 @@
+using System;
 using System.Collections;
 using System.Collections.Generic;
 using System.Reflection;
 using Microsoft.Extensions.Options;
 
 namespace Orleans
 {
-    internal sealed class DefaultOptionsFormatter<T> : IOptionFormatter<T>
-         where T : class, new()
+    internal sealed class DefaultOptionsFormatter<T> : IOptionFormatter<T> where T : class, new()
     {
-        public string Name { get; }
-
-        private readonly T options;
+        private readonly T _options;
 
         public DefaultOptionsFormatter(IOptions<T> options)
         {
-            this.options = options.Value;
-            this.Name = OptionFormattingUtilities.Name<T>();
+            _options = options.Value;
+            Name = OptionFormattingUtilities.Name<T>();
         }
 
         internal DefaultOptionsFormatter(string name, T options)
         {
-            this.options = options;
-            this.Name = OptionFormattingUtilities.Name<T>(name);
+            _options = options;
+            Name = OptionFormattingUtilities.Name<T>(name);
         }
 
+        public string Name { get; }
+
         public IEnumerable<string> Format()
         {
-            var result = new List<string>();
-            foreach (var prop in typeof(T).GetProperties(BindingFlags.Instance | BindingFlags.Public))
+            foreach (var prop in typeof(T).GetProperties())
             {
-                if (prop.GetGetMethod() is object && prop.GetSetMethod() is object)
+                if (!IsFormattableProperty(prop))
                 {
-                    FormatProperty(prop, result);
+                    continue;
+                }
+
+                foreach (var formattedValue in FormatProperty(_options, prop))
+                {
+                    yield return formattedValue;
                 }
             }
+        }
 
-            return result;
+        private static bool IsFormattableProperty(PropertyInfo prop)
```

Comment:
This is the crux of the PR: to check the accessibility of the property/accessors before formatting that property